Xlib/ext/randr.py

```diff
@@ -1186,12 +1267,20 @@ def init(disp, info):
     disp.extension_add_method('display', 'xrandr_get_panning', get_panning)
     disp.extension_add_method('display', 'xrandr_set_panning', set_panning)
 
-    disp.extension_add_event(info.first_event + RRScreenChangeNotify, ScreenChangeNotify)
-     # add RRNotify events (1 event code with 3 subcodes)
-    disp.extension_add_subevent(info.first_event + RRNotify, RRNotify_CrtcChange, CrtcChangeNotify)
-    disp.extension_add_subevent(info.first_event + RRNotify, RRNotify_OutputChange, OutputChangeNotify)
-    disp.extension_add_subevent(info.first_event + RRNotify, RRNotify_OutputProperty, OutputPropertyNotify)
-
-    #disp.extension_add_error(BadRROutput, BadRROutputError)
-    #disp.extension_add_error(BadRRCrtc, BadRRCrtcError)
-    #disp.extension_add_error(BadRRMode, BadRRModeError)
+    # If the server is running RANDR 1.5+, enable 1.5 compatible methods and events
+    version = query_version(disp)
+    if version.major_version == 1 and version.minor_version >= 5:
+        # version 1.5 compatible
+        disp.extension_add_method('window', 'xrandr_get_monitors', get_monitors)
+        disp.extension_add_method('window', 'xrandr_set_monitor', set_monitor)
+        disp.extension_add_method('window', 'xrandr_delete_monitor', delete_monitor)
+
+        disp.extension_add_event(info.first_event + RRScreenChangeNotify, ScreenChangeNotify)
+        # add RRNotify events (1 event code with 3 subcodes)
+        disp.extension_add_subevent(info.first_event + RRNotify, RRNotify_CrtcChange, CrtcChangeNotify)
+        disp.extension_add_subevent(info.first_event + RRNotify, RRNotify_OutputChange, OutputChangeNotify)
+        disp.extension_add_subevent(info.first_event + RRNotify, RRNotify_OutputProperty, OutputPropertyNotify)
+
+        #disp.extension_add_error(BadRROutput, BadRROutputError)
+        #disp.extension_add_error(BadRRCrtc, BadRRCrtcError)
+        #disp.extension_add_error(BadRRMode, BadRRModeError)
```

Comment:
What is the reason to keep this commented code? Please add a comment when it should be uncommented or please remove this code.

Reply:
I was hoping @allfro  would chime in for this and the `pack_value` changes. It seems the error classes referenced weren't defined, I've defined them and uncommented the code